homeassistant/components/nintendo_parental/entity.py

```diff
@@ -0,0 +1,34 @@
+"""Base entity definition for Nintendo Parental."""
+
+from __future__ import annotations
+
+from pynintendoparental import Device
+
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import NintendoUpdateCoordinator
+
+
+class NintendoDevice(CoordinatorEntity[NintendoUpdateCoordinator]):
+    """Represent a Nintendo Switch."""
+
+    def __init__(self, coordinator: NintendoUpdateCoordinator, device: Device) -> None:
+        """Initialize."""
+        super().__init__(coordinator=coordinator)
+        self._device = device
+        self._attr_unique_id = f"{device.device_id}_{self.entity_description.key}"
+        self._attr_has_entity_name = True
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device.device_id)},
+            manufacturer="Nintendo",
+            name=device.name,
+            entry_type=DeviceEntryType.SERVICE,
+            sw_version=device.extra["device"]["firmwareVersion"]["displayedVersion"],
```

Comment:
so its a service, but on the other hand we also show the device version?

Reply:
Yes, a service because this doesn't technically interact with a device, but maybe I'm thinking to "technical" and actually it should be a device, because it does actually represent a device I suppose...

Shall I change to a device?

Reply:
Changed to a device